core/common/src/main/java/alluxio/security/authentication/SaslStreamServerDriver.java

```diff
@@ -81,9 +82,15 @@ public void onNext(SaslMessage saslMessage) {
       // Complete the call from server-side before sending success response to client.
       // Because client will assume authenticated after receiving the success message.
       if (response.getMessageType() == SaslMessageType.SUCCESS) {
+        // Register authorized user with the authentication server.
         mAuthenticationServer.registerChannel(mChannelId,
-            mSaslServerHandler.getAuthenticatedUserInfo(), this,
-            mSaslServerHandler.getSaslServer());
+            mSaslServerHandler.getAuthenticatedUserInfo(), this);
+        // Finished with the handler.
+        try {
+          mSaslServerHandler.close();
```

Comment:
Should we also close the handle when channel is not authenticated successfully?

Reply:
Definitely. Fixed.